src/planner/planner.cpp

```diff
@@ -103,7 +103,9 @@ void Planner::PlanExecute(unique_ptr<SQLStatement> statement) {
 	vector<Value> bind_values;
 	for (idx_t i = 0; i < stmt.values.size(); i++) {
 		ConstantBinder cbinder(*binder, context, "EXECUTE statement");
-		cbinder.target_type = prepared->GetType(i + 1);
+		if (prepared->value_map.count(i + 1)) {
```

Comment:
Should we do a check here? It seems like this should be caught earlier on in the process, as this seems to indicate a mismatch in number of prepared statement parameters, no?

Reply:
Actually this is the case we are fixing!  The problem is not that there are the wrong number of parameters, but that not all of them are used.  The validation happens in the `Bind` call just below.